### ISSUE TITLE
Seeking maintainer(s)

### ISSUE THREAD

**Issue by paralin** [IrcDotNet/IrcDotNet#56]
I've dreamed of having enough spare time to bring this repo up to date for a long time. Unfortunately, both my personal work and my professional work has moved completely away from C#, and I haven't used it in over a year. For this reason I have not had the opportunity to update this package in quite some time.

Looking for one or more community members using this package that would be interested in adopting this package.

**Comment by ransagy** [comment]
While i don't nearly have the time to oversee the entire project, I'd be more than willing to help with small tasks and the like if/when someone steps up to take that position. I've been working with .NET and C# for over a decade now and should be able to help out with short-term changes and quick fixes.

**Comment by izrik** [comment]
I would love to be involved on this project. I probably don't have the requisite experience to be the primary maintainer, but I'm definitely available to submit PR's, etc.

**Comment by paralin** [comment]
Let's do this then. I'll still maintain in terms of reviewing PRs and hitting the final commit and release buttons. Let's set up a working group of community maintainers that also review PRs when possible and in general try to address issues in the code:

 - @ransagy - short term changes and fixes (assigned via issues), code review
 - @izrik - issue fixes and features

I will update this list a bit soon, and try to draft up something more concrete. If you guys want to start submitting PRs I will review them as possible

**Comment by chrboe** [comment]
I'm in a similar position as you @paralin, I haven't worked with C# in a long time and don't use the library anymore, though I'd be more than happy to review some PRs here and there (I am fairly familiar with the codebase after all)

**Comment by paralin** [comment]
@chrboe Great, will add you to the list.

@chrboe @ransagy I'd love if you guys could give the other PR's in the queue reviews/tests as well if possible.

In general we need to fix the CI before we can start merging things. #57

**Comment by paralin** [comment]
@chrboe I know you submitted a few of the pending PRs - will finally get around to getting those in :)

**Comment by alexreg** [comment]
Thanks for making the effort to keep this project well-maintained, @paralin. As someone who hasn't used C# much since handing over this project to you and others, I can sympathise. But it's great if someone can still help maintain it, at least to oversee PRs and the like!

**Comment by txdv** [comment]
I can chip in, I don't code C# professionally currently, but it is still an awesome language that I like to code in my freetime.

**Comment by txdv** [comment]
@paralin @alexreg ? You guys up?

**Comment by alexreg** [comment]
@txdv I don't really do much C# any more, but I'm happy to pass this along to a maintainer who is serious about keeping it up-to-date and the least bug-fixing/reviewing PRs. :-)

**Comment by txdv** [comment]
@alexreg Hey, I am serious! I think I would manage to find more time than currently is being spent.

**Comment by alexreg** [comment]
Oh I know you are, no jokes! @paralin would you be happy to either share or pass on maintenance duties to @txdv?

**Comment by paralin** [comment]
No worries, I'm way too busy working on Go and C/Linux related stuff these days to follow C# :)

**Comment by txdv** [comment]
So what is the protocol now @paralin @alexreg ?

**Comment by alexreg** [comment]
@txdv I've sent you an invitation to the IrcDotNet team. Welcome! If you have any questions on how things should be managed, or releases performed, go first to @paralin and then to me.

**Comment by oreleraki** [comment]
Is there any chance to make this project activate again ? i have some free time to do new coding and maintenance, but i'm not familiar with OS git procedure to take full control yet.

**Comment by alexreg** [comment]
@oreleraki That would be nice. Maybe you could start by submitting a few PRs? The maintainers will hopefully at least be able to review. If that goes well, could potentially make you a maintainer then.

**Comment by oreleraki** [comment]
@alexreg I done it a month ago, with 2 feature and 1 bug PR, but they are still in the Pull Request section, without any review or approval. I can submit more, if i will see any real progress in accepting or at least reviewing them and give comments.

**Comment by alexreg** [comment]
@oreleraki Ah sorry,  I wasn't aware, because I don't actively maintain this project any more or even get PR notifications. (I was the original author and maintainer however.) @txdv, would you be happy to review them?

If he doesn't get around to that in the next few weeks, I suppose I can have a quick look, and if they look good, I'll pass over maintainership to you. Just very busy at the moment, I'm afraid.

**Comment by oreleraki** [comment]
@alexreg, Still no reply, which is too bad we're letting this project to become obsolete.

**Comment by alexreg** [comment]
@oreleraki I’ll review your PRs later today and give you maintainership if I’m happy with them.

**Comment by alexreg** [comment]
@oreleraki Okay. I had a look at your three recent(ish) pull requests. Look good, basically. Just observe the few minor comments I left, and let's also think about adding more tests (probably the testing situation could be better to begin with).

I've gone ahead and granted you "write" privileges. Hopefully that should suffice for now, but happy to increase them later if need be. If you find there's someone else making regular contributions too, let me know, and I can gran them access as well.

Thanks for your desire to maintain this project!

**Comment by oreleraki** [comment]
@alexreg, I dosen't allow me to do any "Merge Pull Request", maybe there is still missing. Also i think i should have targeted the fix to "release" and not master. Additionally, is there anywhere else i can communicate with you in case of questions ? (IRC?)

**Comment by alexreg** [comment]
@oreleraki GitHub says you already have permission to merge pull requests...

You can communicate with me by email or here on GH if you like, though preferably only for the odd thing that really needs my attention, since I don't want to be involved in day-to-day maintenance of this project these days. :-)

**Comment by oreleraki** [comment]
@alexreg I have fixed and aligned the PR and they are ready. but i can't PR approve them in master.
I think you may gave me access to "release" and not "master".

Also, should we do a complete PR from release to master to align ourselves with a clean slake ?

**Comment by alexreg** [comment]
@oreleraki Nope, you have access to all branches. Have you tried simply pushing to master?

The idea is to use `release` as a branch for stabilisations and bug-fixes, but if you wish to adopt a different model, I'm okay with that.

**Comment by oreleraki** [comment]
@alexreg As far as i know Git flow standard are as following:
We have 3 main branches: *master*(constant), *develop*(constant) and *release*(temporary).
*master* is the most stable version release and always production ready.
*develop* most changed branch, which contain all PR from different team members from **features** branches and **bugfix** branches.
*release* spawned from *develop* when all future features and bugfixes are completed, the branch is a candidate for production to become **master**. In this period, testers checks release and every bug that is found, is fixed and the bugfix branch is also PR to develop. when this release branch is ready, a PR from release to master takes place, and master branch is now tagged with the version number.
** There are more to say, but this is the majority of the **git flow** that most organization are using.

Are there any objection to this ?

**Comment by alexreg** [comment]
@oreleraki I'm not a huge fan of Git flow personally, but I see why people like it. (I prefer just version and/or feature branches, I'd say.) But no, no objections, since you'd be maintaining it... as long as you are comfortable.

**Comment by gunnbr** [comment]
@paralin or @alexreg , are there any maintainers left on this project? Like @Felk was a few years ago, I'm also trying write a bot for Twitch, but can't use IrcDotNet since Twitch requires turning on some features through capability negotiation. It looks like PR #39 and #51 are ready to merge and would make IrcDotNet much more useful. Is there anyone around who can do this?

Thank you!

**Comment by paralin** [comment]
@gunnbr merged
